src/storage/s3.rs

```diff
@@ -21,6 +21,19 @@ use tracing::{error, warn};
 const PUBLIC_ACCESS_TAG: &str = "static-cloudfront-access";
 const PUBLIC_ACCESS_VALUE: &str = "allow";
 
+fn err_is_not_found<E>(err: &SdkError<E>) -> bool
+where
+    E: ProvideErrorMetadata,
+{
+    match err {
+        SdkError::ServiceError(err) => {
+            err.raw().status().as_u16() == http::StatusCode::NOT_FOUND.as_u16()
+        }
+        e if e.code() == Some("KeyTooLongError") => true,
```

Comment:
Does minio also return this error in some situations? Is it possible to write a testcase that triggers it?

Reply:
it seems to have it: 
https://github.com/minio/minio/blob/8cd967803cbccaac87bf71123ff86cff9281fc62/cmd/api-errors.go#L196 

let me check if I can trigger it